nix/module.nix

```diff
@@ -30,47 +27,32 @@ in {
         '';
       };
 
-    package = mkOption {
-      type = types.path;
-      default = defaultHyprlandPackage;
-      defaultText = literalExpression ''
-        hyprland.packages.''${pkgs.stdenv.hostPlatform.system}.default.override {
-          enableXWayland = config.programs.hyprland.xwayland.enable;
-          hidpiXWayland = config.programs.hyprland.xwayland.hidpi;
-          inherit (config.programs.hyprland) nvidiaPatches;
-        }
-      '';
-      example = literalExpression "pkgs.hyprland";
+    package = mkPackageOptionMD inputs.packages.${pkgs.system} "hyprland" {};
+
+    finalPackage = mkOption {
```

Comment:
See my other comment for how I set up the `package'`, but as this currently currently stands, you're missing `hidden = true`. Unless you intend to make the `finalPackage` available through `config` for the sake of recursively referencing the binaries in configuration.

Reply:
That's the usecase, indeed.